### ISSUE TITLE
Add an `upload_item_to_s3` method

### ISSUE THREAD

**Issue by pjhartzell** [stac-utils/stac-task#153]
A companion `upload_item_to_s3` method to the existing `upload_item_assets_to_s3` method would be useful for storing a canonical STAC Item to live alongside the assets in S3.

**Comment by jkeifer** [comment]
An issue here is that the item is updated after the task completes. We need to consider if this should just happen automatically. Or if the item updates should be done manually rather than automatically in the background.

Per discussion, we're leaning to making all enrichment/actions be manual for now. Once we get enough and the boilerplate becomes too onerous we can revisit how best to make applying what you need easier.

tl;dr: we want this method. But we need to pull the automatic item updates out of being done automatically in some way so we can call that before calling this function.
